### PR TITLE
sort processor - support rows with duplicate keys

### PR DESCRIPTION
**Pull request by OriHoch**
### reproduction
* use the `sort` processor on rows which have duplicate keys

#### expected
* keep all rows and in case of duplicate keys - keep the rows under the same key in the order in which they appeared

#### actual
* rows with duplicate keys are not yielded and only the last row is yielded for each key

### CODE REVIEW

**Comment by OriHoch** [comment]
not sure what are the implications on performance, but I tested it with some real data and it seems to work well

**Comment by OriHoch** [comment]
after some testing I changed to a better implementation - append a row number to each key

**Comment by OriHoch** [comment]
```
knesset-data-committees$ time dpp run ./join-attendance-data
INFO    :Main                            :Skipping redis connection, host:None, port:6379
INFO    :Main                            :8e5b24fa RUNNING ./join-attendance-data
INFO    :Main                            :8e5b24fa Collecting dependencies
INFO    :Main                            :8e5b24fa Running async task
INFO    :Main                            :8e5b24fa Waiting for completion
INFO    :Main                            :8e5b24fa Async task starting
INFO    :Main                            :8e5b24fa Searching for existing caches
INFO    :Main                            :8e5b24fa Building process chain:
INFO    :Main                            :- load_resource
INFO    :Main                            :- load_resource
INFO    :Main                            :- stream_remote_resources
INFO    :Main                            :- join
INFO    :Main                            :- sort
INFO    :Main                            :- dump.to_path
INFO    :Main                            :- (sink)
INFO    :Main                            :sort: INFO    :Using leveldb for joining
INFO    :Main                            :join: INFO    :Using leveldb for joining
INFO    :Main                            :load_resource: INFO    :Processed 34159 rows
INFO    :Main                            :8e5b24fa DONE /home/ori/datapackage-pipelines/datapackage_pipelines/specs/../lib/load_resource.py
INFO    :Main                            :load_resource: INFO    :Processed 853509 rows
INFO    :Main                            :8e5b24fa DONE /home/ori/datapackage-pipelines/datapackage_pipelines/specs/../lib/load_resource.py
INFO    :Main                            :stream_remote_resources: INFO    :Processed 853509 rows
INFO    :Main                            :8e5b24fa DONE /home/ori/datapackage-pipelines/datapackage_pipelines/specs/../lib/stream_remote_resources.py
INFO    :Main                            :join: INFO    :Processed 819350 rows
INFO    :Main                            :8e5b24fa DONE /home/ori/datapackage-pipelines/datapackage_pipelines/specs/../lib/join.py
INFO    :Main                            :dump.to_path: INFO    :Dumped 10000 rows
...
...
INFO    :Main                            :dump.to_path: INFO    :Dumped 810000 rows
INFO    :Main                            :sort: INFO    :Processed 819350 rows
INFO    :Main                            :8e5b24fa DONE /home/ori/datapackage-pipelines/datapackage_pipelines/specs/../lib/sort.py
INFO    :Main                            :dump.to_path: INFO    :Processed 819350 rows
INFO    :Main                            :8e5b24fa DONE /home/ori/datapackage-pipelines/datapackage_pipelines/specs/../lib/dump/to_path.py
INFO    :Main                            :8e5b24fa DONE /home/ori/datapackage-pipelines/datapackage_pipelines/manager/../lib/internal/sink.py
INFO    :Main                            :DONE V ./join-attendance-data {'.dpp': {'out-datapackage-url': 'data/attendance-joined-data/datapackage.json'}, 'bytes': 209199933, 'count_of_rows': 819350, 'dataset_name': '_', 'hash': 'c02f4adc58b0974c1ac82641c0fa8e25'}
INFO    :Main                            :RESULTS:
INFO    :Main                            :SUCCESS: ./join-attendance-data {'bytes': 209199933, 'count_of_rows': 819350, 'dataset_name': '_', 'hash': 'c02f4adc58b0974c1ac82641c0fa8e25'}

real	4m33.518s
user	9m29.923s
sys	0m19.710s
```

File: datapackage_pipelines/lib/sort.py [hunk 1]
```diff
@@ -20,8 +20,8 @@ def __call__(self, row):
 
 def sorter(resource):
     db = KVStore()
-    for row in resource:
-        key = key_calc(row)
+    for row_num, row in enumerate(resource):
+        key = key_calc(row) + "-{}".format(row_num)
```
Comment (akariv) [inline datapackage_pipelines/lib/sort.py]: in a the scenario where rows 999 and 1000 have the same key, row '1000' will come before '999'

Comment (akariv) [reply datapackage_pipelines/lib/sort.py]: using `{:08x}` instead could do the trick

Comment (OriHoch) [reply datapackage_pipelines/lib/sort.py]: do you have a good solution for this? I can add some padding, but I think it unnecessarily complicates things.

I'm don't think guaranteeing that rows with duplicate keys are sorted within the key is a priority

maybe it's better to just say that rows with duplicate keys are unsorted (within the duplicated key)?

Comment (akariv) [reply datapackage_pipelines/lib/sort.py]: @OriHoch ^ proposed solution above

Comment (OriHoch) [reply datapackage_pipelines/lib/sort.py]: @akariv fixed

**Pull request merged**